js/components/MilkdropWindow/PresetOverlay.js

```diff
@@ -49,103 +102,83 @@ class PresetOverlay extends React.Component {
-            position: "absolute",
-            top: 0,
-            left: 0,
-            color: "white",
-            background: "rgba(0.33, 0.33, 0.33, 0.33)"
-          }}
-        >
-          <span>Loading presets</span>
-        </div>
-      );
-    }
+  _renderList() {
+    const { presetKeys, currentPreset, height, width } = this.props;
+    const { selectedListIndex } = this.state;
 
-    // display highlighted preset in the middle if possible
-    const numPresets = this.props.presetKeys.length;
-    let presetListLen = Math.floor(this.props.height / 20);
-    presetListLen = Math.min(Math.max(presetListLen, 3), numPresets);
-    presetListLen = presetListLen % 2 ? presetListLen : presetListLen - 1;
-    const halfPresetListLen = Math.floor(presetListLen / 2);
-    let startIdx = Math.max(this.state.presetIdx - halfPresetListLen, -1);
-    let endIdx = Math.min(startIdx + presetListLen, numPresets);
-    if (endIdx >= numPresets) {
-      startIdx = Math.max(endIdx - presetListLen, -1);
-      endIdx = Math.min(startIdx + presetListLen, numPresets);
-    }
-    startIdx = Math.max(startIdx, 0); // ensure startIdx >= 0 after endIdx is calculated
-    const presets = this.props.presetKeys.slice(startIdx, endIdx);
-    const presetElms = presets.map((presetName, i) => {
-      let color;
-      if (i + startIdx === this.props.currentPreset) {
-        if (i + startIdx === this.state.presetIdx) {
-          color = "#FFCC22";
-        } else {
-          color = "#CCFF03";
-        }
-      } else if (i + startIdx === this.state.presetIdx) {
-        color = "#FF5050";
-      } else {
-        color = "#CCCCCC";
-      }
-      return (
-        <li key={i} style={{ color }}>
-          {presetName}
-        </li>
-      );
-    });
+    const maxVisibleRows = Math.floor((height - HEIGHT_PADDING) / ENTRY_HEIGHT);
+    const rowsToShow = Math.floor(maxVisibleRows * 0.75); // Only fill 3/4 of the screen.
+    const [startIndex, endIndex] = getRangeCenteredOnIndex(
+      this._maxListIndex() + 1, // Add one to convert an index to a length
+      rowsToShow,
+      selectedListIndex
+    );
 
-    if (this.state.presetIdx - halfPresetListLen < 0) {
+    const presetElms = [];
+    for (let i = startIndex; i <= endIndex; i++) {
+      const presetIndex = this._presetIndexFromListIndex(i);
+      const isSelected = i === selectedListIndex;
+      const isCurrent = presetIndex === currentPreset;
       let color;
-      if (this.state.presetIdx === -1) {
-        color = "#FF5050";
+      if (isSelected) {
+        color = isCurrent ? "#FFCC22" : "#FF5050";
       } else {
-        color = "#CCCCCC";
+        color = isCurrent ? "#CCFF03" : "#CCCCCC";
       }
-      presetElms.unshift(
-        <li key={"localDir"} style={{ color }}>
-          Load Local Directory
+      presetElms.push(
+        <li key={i} style={{ color, lineHeight: `${ENTRY_HEIGHT}px` }}>
+          {i === 0 ? "Load Local Directory" : presetKeys[presetIndex]};
```

Comment:
I think the semi at the end here is accidental?

Reply:
Good catch! Thanks so much for taking a look.